Paintroid/src/main/java/org/catrobat/paintroid/presenter/LayerPresenter.kt

```diff
@@ -167,21 +167,23 @@ class LayerPresenter(
     }
 
     override fun hideLayer(position: Int) {
-        val destinationLayer = getDestinationLayer(position, false)
-        drawingSurface?.refreshDrawingSurface()
-        if (model.currentLayer == destinationLayer) {
-            defaultToolController?.switchTool(ToolType.HAND, false)
-            bottomNavigationViewHolder?.showCurrentTool(ToolType.HAND)
+        getDestinationLayer(position, false)?.let { layer ->
+            drawingSurface?.refreshDrawingSurface()
```

Comment:
```suggestion
        drawingSurface?.refreshDrawingSurface()
        getDestinationLayer(position, false)?.let { layer ->
```
shouldn't `drawingSurface?.refreshDrawingSurface()` be called irrespective of whether `getDestinationLayer(position, false)` is null or not, the logic seems to be changed here 🤔, not very sure though

Reply:
no, we do not want to do anything here i think. We could not hide the layer since we could not fetch the destination layer (for some unknown concurrency reason) and therefore we do not want to interact and hope the user clicks again and hope that the (unknown) reason does not occure again.

Do you aggree? Not 100% sure either, defintely had to think about this.

Reply:
hmm.. makes sense 👍🏻. @ThorstenBandel could you please just take a look at it once, I think you'll have a better idea about this?

Reply:
i know changed it according to your suggestion, shouldnt matter either way - but just in case ht drawingSurface receives a lock from the outside, it will be cleared in case of failure.